go/lib.h

```diff
@@ -46,10 +50,18 @@ typedef void (*FailureCallback)(uintptr_t channel_address,
  *
  * The returned `ConnectionResponse` should be manually freed by calling `free_connection_response`, otherwise a memory leak will occur. It should be freed whether or not an error occurs.
  *
+ * `connection_request_bytes` is an array of bytes that will be parsed into a Protobuf `ConnectionRequest` object.
+ * `connection_request_len` is the number of bytes in `connection_request_bytes`.
+ * `success_callback` is the callback that will be called in the case that the Redis command succeeds.
+ * `failure_callback` is the callback that will be called in the case that the Redis command fails.
+ *
  * # Safety
  *
- * * `connection_request_bytes` must point to `connection_request_len` consecutive properly initialized bytes.
- * * `connection_request_len` must not be greater than `isize::MAX`. See the safety documentation of [`std::slice::from_raw_parts`](https://doc.rust-lang.org/std/slice/fn.from_raw_parts.html).
+ * * `connection_request_bytes` must point to `connection_request_len` consecutive properly initialized bytes. It should be a well-formed Protobuf `ConnectionRequest` object. The array must be allocated on the Golang side and subsequently freed there too after this function returns.
```

Comment:
the convention is to state whether the caller or the callee release a pointer, in order to be clear.